core/src/test/java/org/infinispan/distribution/ZeroCapacityNodeTest.java

```diff
@@ -149,6 +156,54 @@ private void createCache(ConfigurationBuilder cb, String cacheName) {
       waitForClusterToForm(cacheName);
    }
 
+   public void testZeroCapacityFactorNodeStartsFirst() throws Exception {
+      ConfigurationBuilder cb = new ConfigurationBuilder();
+      cb.clustering().cacheMode(CacheMode.DIST_SYNC);
+      cb.clustering().hash().numSegments(NUM_SEGMENTS);
+      cb.clustering().remoteTimeout(100);
+
+      ConfigurationBuilder cbZero = new ConfigurationBuilder();
+      cbZero.clustering().cacheMode(CacheMode.DIST_SYNC);
+      cbZero.clustering().hash().numSegments(NUM_SEGMENTS);
+      cbZero.clustering().hash().capacityFactor(0f);
+      cbZero.clustering().remoteTimeout(100);
+
+      String cacheName = "testCache";
+      Future<Cache<Object, Object>> zeroCapacityNodeFuture =
+            fork(() -> zeroCapacityNode.createCache(cacheName, cb.build()));
+      Future<Cache<Object, Object>> node1Future =
+            fork(() -> node1.createCache(cacheName, cbZero.build()));
+
+      TestingUtil.sleepThread(10);
```

Comment:
Can we not make use of the `TimeService` here to avoid waiting for 10 seconds?

Reply:
The wait is just 10ms, not 10s, but I'll give it a try.

Reply:
> The wait is just 10ms, not 10s, but I'll give it a try.

Argh I totally missed that :see_no_evil:. Thanks for updating.